hyperspy/_components/expression.py

```diff
@@ -118,14 +147,33 @@ class docstring.
             self.__doc__ = _CLASS_DOC % (
                 name, sympy.latex(_parse_substitutions(expression)))
 
-    def function(self, x):
-        return self._f(x, *[p.value for p in self.parameters])
-
-    def compile_function(self, module="numpy"):
+    def compile_function(self, module="numpy", position=False):
         import sympy
         from sympy.utilities.lambdify import lambdify
         expr = _parse_substitutions(self._str_expression)
-
+        # Extract x
+        x, = [symbol for symbol in expr.free_symbols if symbol.name == "x"]
+        # Extract y
+        y = [symbol for symbol in expr.free_symbols if symbol.name == "y"]
+        self._is2D = True if y else False
+        if self._is2D:
+            y = y[0]
+        if self._is2D and self._add_rotation:
+            if position:  # Rotate around the "center" of the function
+                rotx = sympy.sympify(
+                    "{0} + (x - {0}) * cos(rotation_angle) - (y - {1}) *"
+                    " sin(rotation_angle)"
+                    .format(*position))
+                roty = sympy.sympify(
+                    "{1} + (x - {0}) * sin(rotation_angle) + (y - {1}) *"
+                    "cos(rotation_angle)"
+                    .format(*position))
+            else:  # Rotate around the origin
+                rotx = sympy.sympify(
+                    "x * cos(rotation_angle) - y * sin(rotation_angle)")
+                roty = sympy.sympify(
+                    "x * sin(rotation_angle) + y * cos(rotation_angle)")
+            expr = expr.subs({"x": rotx, "y": roty}, simultaneous=False)
```

Comment:
I believe this if/else play is unneeded, and could be written as
```python
position = position or (0,0)
# proceed to use the "if position" code
```
when `sympy.sympify` parses the string, it also simplifies it - so the zeroes vanish anyway.

Also the default in the function is `False`, but it would never be so - always either a user-given value, or a `None` from `__init__` default is passed